client/app/hosting/database/private-activation/database-private-activation.controller.js

```diff
@@ -1,52 +1,71 @@
-angular.module("App").controller("HostingDatabasePrivateActiveCtrl", ($scope, $rootScope, $stateParams, HostingDatabase, Alerter, Hosting) => {
-    "use strict";
+angular.module("App").controller(
+    "HostingDatabasePrivateActiveCtrl",
+    class HostingDatabasePrivateActiveCtrl {
 
-    $scope.hosting = $scope.currentActionData;
+        constructor ($scope, $rootScope, $stateParams, HostingDatabase, Alerter, PrivateDatabase) {
+            this.$scope = $scope;
+            this.$rootScope = $rootScope;
+            this.$stateParams = $stateParams;
+            this.HostingDatabase = HostingDatabase;
+            this.Alerter = Alerter;
+            this.PrivateDatabase = PrivateDatabase;
 
-    $scope.datas = {
-        versions: []
-    };
-    $scope.loaders = {
-        versions: false
-    };
-    $scope.choice = {
-        ram: $scope.hosting.offerCapabilities.privateDatabases.length === 1 ? $scope.hosting.offerCapabilities.privateDatabases[0] : null,
-        version: null
-    };
+            this.host = this.$scope.currentActionData;
+            this.versions = [];
+            this.loaders = {
+                versions: false
+            };
+            this.choice = {
+                ram: this.host.offerCapabilities.privateDatabases.length === 1 ?
+                       _.first(this.host.offerCapabilities.privateDatabases) : null,
+                version: null
+            };
 
-    function init () {
-        if (!$scope.loaders.versions) {
-            $scope.loaders.versions = true;
-            Hosting.getModels()
-                .then((models) => {
-                    $scope.datas.versions = models.models["hosting.PrivateDatabase.OrderableVersionEnum"].enum;
-                    $scope.choice.version = $scope.datas.versions.length === 1 ? $scope.datas.versions[0] : null;
-                })
-                .catch((err) => {
-                    Alerter.alertFromSWS($scope.tr("hosting_dashboard_database_versions_error", [$scope.entryToDelete]), _.get(err, "data", err), $scope.alerts.main);
-                })
-                .finally(() => {
-                    $scope.loaders.versions = false;
-                });
+            this.$scope.activateDatabase = () => this.activateDatabase();
         }
-    }
 
-    $scope.activeDatabase = function () {
-        if (!$scope.loaders.versions) {
-            $scope.loaders.versions = true;
-            HostingDatabase.activeDatabasePrivate($stateParams.productId, $scope.choice.ram.quota.value, $scope.choice.version)
+        $onInit () {
+            if (!this.loaders.versions) {
+                this.loaders.versions = true;
+                this.PrivateDatabase.getAvailableOrderCapacities("classic")
+                    .then((capacities) => {
+                        this.versions = _.get(capacities, "version");
```

Comment:
You're trying to prevent a `TypeError` by not getting `version` if it doesn't exist and then you call `length` on `this.version`, fully expecting `this.versions` to be an array.  Please check the types more carefully :)